Dashboard/app/js/lib/views/surveys/survey-details-views.js

```diff
@@ -493,51 +493,28 @@ FLOW.QuestionGroupItemView = FLOW.View.extend({
     FLOW.selectedControl.set('selectedForMoveQuestionGroup', null);
   },
 
-  /*
-   *  Request question group and check whether copying is completed on the server side
-   *  then load questions for that group.
-   */
-  ajaxCall: function(qgId){
-      var self = this;
-      $.ajax({
-          url: '/rest/question_groups/' + qgId,
-          type: 'GET',
-          success: function(data) {
-            if (data.question_group.status == "READY") {
-                // reload this question group the Ember way, so the UI is updated
-                FLOW.questionGroupControl.getQuestionGroup(self.content.get('keyId'));
-                // load the questions inside this question group
-                FLOW.questionControl.populateQuestionGroupQuestions(self.content.get('keyId'));
-            }
-          },
-          error: function() {
-            console.error("Error in checking ready status survey group copy");
-          }
-      });
-  },
-
-  qgCheckScheduled: false,
-
   // cycle until our local question group has an id
   // when this is done, start monitoring the status of the remote question group
   pollQuestionGroupStatus: function(){
       var self = this, qgQuery = null;
       if (this.get('amCopying')) {
-        if (!this.get('qgCheckScheduled')) {
-          this.set('qgCheckScheduled', true);
           qgQuery = setInterval(function () {
               // if the question group has a keyId, we can start polling it remotely
               if (self.content && self.content.get('keyId')) {
+                // we have an id and can start polling remotely
                 if (self.content.get('status') == "READY") {
-                  self.set('qgCheckScheduled', false);
+                  //load new group's questions and clear interval
+                  FLOW.store.findQuery(FLOW.Question, {
```

Comment:
One thing I am wondering about here.. The question group polling starts when a question group has just been created, no?  If that's the only case where polling is done, then it should not yet have questions added to it, so I was just wondering how come we are trying to get its questions.

Reply:
When we copy a question group, the newly created group's status is set to copying which the `pollQuestionGroupStatus` function is set to observe. We when set up an interval to get all question groups every 5s and check if the copied question group's status has changed to `READY` in which case we pull its questions and clear the interval. So we only pull a question group's questions once the group is ready